theano/tensor/basic.py

```diff
@@ -4078,6 +4078,79 @@ def R_op(self, inputs, eval_points):
             return [None]
         return self.make_node(*eval_points).outputs
 
+    def c_code_cache_version(self):
+        return (1,)
+
+    def c_code(self, node, name, inputs, outputs, sub):
+        x, = inputs
+        out, = outputs
+        outdim = self.outdim
+        fail = sub['fail']
+        return """
+        if (%(outdim)s == PyArray_NDIM(%(x)s))
+        {
+            if (NULL != %(out)s && %(x)s != %(out)s)
+                Py_XDECREF(%(out)s);
+
+            %(out)s = %(x)s;
```

Comment:
I believe that incorrect reference counting is the reason for the segfaults. I added `Py_XINCREF(%(x)s);` before `%(out)s = %(x)s;` locally and the segfaults are gone, but I'm not 100% sure if that's the right thing to do here. How do I handle reference counting with views?